components/interface/TreeWidget.js

```diff
@@ -383,47 +382,97 @@ export default class TreeWidget extends React.Component {
     });
   }
 
+  findRoot (nodes) {
+    let min = nodes[0].id;
+    for ( let i = 1; i < nodes.length; i++) {
+      if(nodes[i].id < min) {
+        min = nodes[i].id;
+      }
+    }
+    return min;
+  }
+
   componentWillMount () {
     if (window.templateID !== undefined) {
       this.initTree(window.templateID);
+      document.addEventListener('mousedown', this.monitorMouseClick, false);
     }
   }
 
+  componentWillReceiveProps () {
+    document.removeEventListener('mousedown', this.monitorMouseClick, false);
```

Comment:
@ddelpiano why here and not in WillUnmount?

Reply:
already fixed in the latest